File: Dart/factorial.dart
int factorial(int n) => n < 2 ? 1 : n * factorial(n - 1);

// int factorial(int n) {
//     return n < 2 ? 1 : n * factorial(n - 1);
// }

main() {
    print(factorial(1));
    print(factorial(2));

    print(factorial(5));
    print(factorial(6));
}

File: Zig/factorial.zig
const warn = @import("std").debug.warn;

fn factorial(n: i32) i32 {
    if (n < 2) return 1
    else return n * factorial(n - 1);
}

pub fn main() void {
    warn("{}\n", factorial(1)); // 1
    warn("{}\n", factorial(5)); // 120
    warn("{}\n", factorial(7)); // 5040
}